pkg/controller/common/common.go

```diff
@@ -0,0 +1,70 @@
+package common
```

Comment:
I think packages named common, utils, etc., tend to be vague, as they imply shared logic without a clearly defined domain or separation of concerns.

In this file, the main struct is `CommonReconciler`, but it's not clear what exactly is being reconciled. The struct also mixes responsibilities: as it's constructing and returning heterogeneous components like backup.PBM, mongo.Client, a scheme, and a k8s client.

To improve clarity and maintainability, I'd suggest:

- Keeping the scheme and the Kubernetes client in `ReconcilePerconaServerMongoDB`, and having related function with receivers of type `ReconcilePerconaServerMongoDB`.

- Splitting out PBM-related logic into a dedicated PBM factory/service.

- Doing the same for the MongoClientProvider.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1926/commits/9bf2482619d20baa3636cf361fe006f019b8684b